econml/ortho_forest.py

```diff
@@ -252,96 +320,58 @@ def const_marginal_effect(self, X):
         Theta : matrix , shape (n, d_t)
             Constant marginal CATE of each treatment for each sample.
         """
+        # TODO: Check performance
+        return np.asarray(self._predict(X))
+
+    def _predict(self, X, stderr=False):
         if not self.model_is_fitted:
             raise NotFittedError('This {0} instance is not fitted yet.'.format(self.__class__.__name__))
         X = check_array(X)
-        results = Parallel(n_jobs=self.n_jobs, verbose=3, backend='threading')(
-            delayed(self._pointwise_effect)(X_single) for X_single in X)
-        # TODO: Check performance
-        return np.asarray(results)
 
-    def _get_inference_options(self):
-        # Override the CATE inference options
-        # Add blb inference to parent's options
-        options = super()._get_inference_options()
-        options.update(blb=BLBInference)
-        options.update(auto=BLBInference)
-        return options
-
-    def _pointwise_effect(self, X_single, stderr=False):
-        """Calculate the effect for a one data point with features X_single.
-
-        Parameters
-        ----------
-        X_single : array-like, shape (d_x, )
-            Feature vector that captures heterogeneity for one sample.
-
-        stderr : boolean (default=False)
-            Whether to calculate the covariance matrix via bootstrap-of-little-bags.
-        """
+        def pw_effect_inputs(mask_w1, mask_w2, w_nonzero, split_inds, slice_weights_list):
```

Comment:
This can be a private helper function outside of `_predict` since it's not using anything from `_predict` so it doesn't have to be redefined every time we call `_predict`. Alternatively, this can be grouped with `_nonzero_weights` since `_nonzero_weights` is only used inside this function.

Reply:
good point! I just merged with nonzero weights